apps/gallery/js/ImageEditor.js

```diff
@@ -775,6 +783,7 @@ function saveEditedImage() {
  */
 function ImageEditor(imageBlob, container, edits, ready, croponly) {
   this.imageBlob = imageBlob;
+  this.imageExif = edits ? edits.EXIF : null;
```

Comment:
I'm sort of confused now. I don't think you need the EXIF until the user has completed their edits. And I thought that the code above that reads the exif was there to be reading in the background. But I think it is likely that it won't be done by the time that this constructor is called.  So can you just wait and look at edits.EXIF later on when you need it instead of using this new variable?